outputs.tf

```
@@ -9,8 +9,8 @@ output "fqdn" {
 }
 
 output "zone_id" {
-  value       = google_dns_managed_zone.this.id
-  description = "string ||| Zone id in format projects/{{project}}/managedZones/{{name}}"
+  value       = data.ns_workspace.this.block_ref
```

```suggestion
  value       = google_dns_managed_zone.this.name
```

Using above will make it clear to others what is happening.
If the name changes to not use `block_ref`, this is guaranteed to be correct as well.